src/Lecture4.hs

```diff
@@ -256,8 +351,16 @@ the file doesn't have any products.
 🕯 HINT: Have a look at 'mapMaybe' function from 'Data.Maybe' (you may need to import it).
 -}
 
+rowsToNonEmptyList :: [Row] -> Maybe (NonEmpty Row)
+rowsToNonEmptyList []     = Nothing
+rowsToNonEmptyList (x:xs) = Just (x :| xs)
```

Comment:
This is actually a standard function `nonEmpty` from the `Data.List.NonEmpty` module so you don't need to implement your own `rowsToNonEmptyList` 🙂